hs_tools_resource/models.py

```diff
@@ -230,9 +235,57 @@ def remove(cls, element_id):
 class ToolIcon(AbstractMetaDataElement):
     term = 'ToolIcon'
     value = models.CharField(max_length=1024, blank=True, default="")
+    data_url = models.TextField(blank=True, default="")
+
+    @classmethod
+    def _validate_tool_icon(cls, url):
+        try:
+            response = requests.get(url, verify=False)
+        except Exception:
+            raise ValidationError("Failed to read data from given url.")
```

Comment:
@zhiyuli It'd be good to append the exception message into ValidationError message